streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -735,6 +729,135 @@ public void shouldReturnAllStores() {
         }
     }
 
+    @Test
+    public void shouldReturnCorrectStoreTypeOnly() {
+        final String keyValueStoreName = "keyValueStore";
+        final String timestampedKeyValueStoreName = "keyValueTimestampStore";
+        final String windowStoreName = "windowStore";
+        final String timestampedWindowStoreName = "windowTimestampStore";
+        final String sessionStoreName = "sessionStore";
+        final String globalKeyValueStoreName = "globalKeyValueStore";
+        final String globalTimestampedKeyValueStoreName = "globalKeyValueTimestampStore";
```

Comment:
one small nit: should this test include in-memory stores?

Reply:
Sure. Why not.